src/formatoptions.jl

```diff
@@ -0,0 +1,16 @@
+const default_format_options = (4, 100)
+
+struct FormatOptions <: JuliaFormatter.AbstractStyle
+    indent::Int
+    margin::Int
+end
+FormatOptions() = FormatOptions(default_format_options...)
+
+FormatOptions(options::Vararg{Any,length(default_format_options)}) =
+    FormatOptions(something.(options, default_format_options)...)
+
+JuliaFormatter.getstyle(x::FormatOptions) = x
+
+# All functions that don't have a dispatch defined for FormatOptions
+# fallback to the definition for JuliaFormatter.DefaultStyle. This is
+# how we customize the behavior of the formatter.
```

Comment:
I would like to first make sure whether we want to go with JuliaFormatter.jl before giving actual review, but here I also want to comment a bit about configurations here:
Actually with [the `.JuliaFormatter.toml` configuration file](https://domluna.github.io/JuliaFormatter.jl/dev/config/), we don't need to handle by ourselves -- all what we need to do is to search for a configuration file and pass those configurations there like [Juno does](https://github.com/JunoLab/Atom.jl/blob/75eebb773ec222368801968ff5d6ae1e083c4fbf/src/formatter.jl#L11).
Then we get free from tasks to sync with these frontend configuration handling and actual formatter options (in other word we won't need something like [this PR](https://github.com/julia-vscode/LanguageServer.jl/pull/756) anymore.

Reply:
Yup, a config file is also how other prominent linters (e.g. ESLint) work and much more sensible than putting a bunch of options into every editor's UI.

Reply:
I think it's still valuable to provide configuration at the editor level so that formatting files always works as the user expects without having to create a `.JuliaFormatter.toml` file for one-off scripts (if e.g. the user prefers 2-space tabs as used in the DifferentialEquations codebase). I agree that we should hook this up so that a `.JuliaFormatter.toml` file overrides language server configuration.

Reply:
Yes, but imho that should only happen for "common" options like indentation (i.e. options that are shared by all languages).

Reply:
How about this: we make the VS Code config settings scope `application`, so they can _only_ be configured in user settings, but not as workspace or folder settings. I think what we should try to avoid is to have two different ways to configure folder specific settings, and I think the separate `.JuliaFormatter.toml` story then makes more sense than a VS Code specific configuration setting with `window` scope.

I think in terms of implementations this would mean that a `.JuliaFormatter.toml` would always take precedence (if it exists) over any config settings that the LS receives from the client, right?

What is a lot less clear to me is which config settings we should have in the client... Minimally I guess the option to specify a style? But I don't really see a good reason _not_ to allow all the other settings to be specified as well...